subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/internal/AbstractLocator.java

```diff
@@ -34,18 +34,24 @@ protected AbstractLocator(ExecActionFactory execActionFactory) {
     protected abstract List<String> getXcrunFlags();
 
     public File find() {
-        if (cachedLocation == null) {
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            ExecAction execAction = execActionFactory.newExecAction();
-            execAction.executable("xcrun");
-            execAction.workingDir(System.getProperty("user.dir"));
-            execAction.args(getXcrunFlags());
-            execAction.setStandardOutput(outputStream);
-            execAction.execute().assertNormalExitValue();
-
-            cachedLocation = new File(outputStream.toString().replace("\n", ""));
+        File result = cachedLocation;
+        if (result == null) {
```

Comment:
we shouldn't use double checked locking. Just wrap it all in a synchronized block and get rid of the volatile and the extra check. The JVM takes care of making this fast. Or fast enough until the profiler tells us it's the most important thing to improve.